### PR TITLE
Performance improvements and fixes of previous additions

### PR DESCRIPTION
**Pull request by hulpke**
Collecting the next set of additions for merging.

### CODE REVIEW

File: lib/permdeco.gi [hunk 1]
```diff
@@ -51,8 +51,12 @@ local   pcgs,r,hom,A,iso,p,i,depths,ords,b,mo,pc,limit,good,new,start,np;
...
       p:=i[Length(i)][2]; # best split
-      depths:=Concatenation(Filtered(depths,x->x<=A),[p],
-	        Filtered(depths,x->x>A));
+      if not p in depths then
+	depths:=Concatenation(Filtered(depths,x->x<=A),[p],
+		  Filtered(depths,x->x>A));
+      else
+	good:=false;
+      fi;
     else
```
Comment (hulpke) [inline lib/permdeco.gi]: I have no idea what is displayed here. In my editor there is perfect indentation.

Comment (fingolfin) [inline lib/permdeco.gi]: The problem here is that you are mixing tabs and spaces. As a result, the only way to view that code correctly is in an editor with tab width set to 8. Unfortunately, other parts of the code require a tab width of 4 to display correctly... The only sane solution is to use either tabs or spaces, but not to mix them.

**Comment by ChrisJefferson** [comment]
Hi @hulpke . Will this be ready for merge soon? If not, do you mind if I pull the fixes out and merge them separately?

**Comment by hulpke** [comment]
@ChrisJefferson 
If you want to merge, please do so. I rebased on the current Master. (It got pushed aside because of the massive changes in master b/c hpcgap).

**Comment by alex-konovalov** [comment]
@hulpke added "not for release notes" label assuming that essential previous additions are already described there. If something here is still worth mentioning, please let me know.

**Pull request merged**